network/builder/src/builder.rs

```diff
@@ -327,23 +322,24 @@ impl NetworkBuilder {
             None
         };
 
-        // union pubkeys from addrs with pubkeys directly in config
-        let addr_pubkeys_iter = seed_addrs.iter().map(|(peer_id, addrs)| {
-            let pubkey_set: HashSet<_> = addrs
-                .iter()
-                .filter_map(NetworkAddress::find_noise_proto)
-                .collect();
-            (*peer_id, pubkey_set)
-        });
-        for (peer_id, pubkey_set) in addr_pubkeys_iter {
-            seed_pubkeys.entry(peer_id).or_default().extend(pubkey_set);
-        }
+        // Merge pubkeys that may be in the seed addresses
+        // TODO: This seems like a poor choice to hide known keys in the addresses, and possibly even problematic, it's kept for backwards compatibility
```

Comment:
maybe you could clarify this point? in my mind, the pubkey set in the config is purely a backup; 99% of the time you just want to describe your seeds using a network address, which you can use to dial out to your seeds.

Reply:
I had forgot we merged the pubkeys and the addresses in the ValidatorConfig.

I just thought it was weird to have 2 opposing paradigms, but I now realize I basically brought back an old one :/